### PR TITLE
Wait on test workspace removal before running next test

### PR DESCRIPTION
**Pull request by dmytro-ndp**
### What does this PR do?
It prevents situation when next test starts execution before previous test workspace is actually removed.

### What issues does this PR fix or reference?
#11862

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
ci-test

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/AbstractTestWorkspaceServiceClient.java [hunk 1]
```diff
@@ -120,6 +121,27 @@ public void delete(String workspaceName, String userName) throws Exception {
...
+    WaitUtils.waitSuccessCondition(
+        () -> {
+          try {
+            LOG.info(
+                "Check workspace name='{}', id='{}', username='{}' existence",
+                workspaceName,
+                workspace.getId(),
+                userName);
+            return !exists(workspaceName, userName);
+          } catch (Exception e) {
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/AbstractTestWorkspaceServiceClient.java:128]: Is it possible to set some particular exception?

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/AbstractTestWorkspaceServiceClient.java (hunk 1 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/AbstractTestWorkspaceServiceClient.java:128]: it is thrown by `exists()` itself

**Review by musienko-maxim — APPROVED** [review]

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1151/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1151//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11870
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Pull request merged**